docs/SRS-Volere/SRS.tex

```diff
@@ -132,7 +133,126 @@ \subsection{Individual Product Use Cases (PUC's)}
+\textbf{Description:} The system must check each product found on a user-submitted receipt and update the database with relevant purchase information.\\
+\textbf{Rationale:} This requirement ensures that the application accurately records and maintains the user's spending history, which is essential for budgeting and spending analysis.\\
+\textbf{Fit Criterion:} The system should successfully update the database with purchase information, including date, time, location, and purchase price for each product on the receipt.\\
+
+\medskip
+\noindent \textbf{Requirement ID:} FR8\\
+\textbf{Use Case:} 11, 12\\
+\textbf{Description:} The system must analyze the user's purchase history and compare it to the budgeting preferences the user has submitted. It should suggest items to stop purchasing or recommend cheaper grocery alternatives when the user exceeds their budget.\\
+\textbf{Rationale:} This requirement supports the application's goal of helping users manage their expenses by providing personalized budgeting suggestions.\\
+\textbf{Fit Criterion:} The system should provide relevant budgeting suggestions based on the user's purchase history and budgeting preferences.\\
+
+\medskip
+\noindent \textbf{Requirement ID:} FR9\\
+\textbf{Use Case:} 12\\
+\textbf{Description:} When a recently purchased item by the user is found at a lower price, the system should prompt the user with information about the item, its price, and the location where it is available at a lower cost.\\
+\textbf{Rationale:} This requirement enhances user experience by helping users save money by identifying opportunities for cost savings.\\
+\textbf{Fit Criterion:} The system should prompt the user with information about cheaper alternatives when available within the user's location preferences.\\
+
+\medskip
+\noindent \textbf{Requirement ID:} FR10\\
+\textbf{Use Case:} 10\\
+\textbf{Description:} The system must retrieve and display a visual representation of the user's purchase history, including details on items purchased and their respective timestamps.\\
+\textbf{Rationale:} This requirement enables users to monitor their spending habits and make informed financial decisions by providing a clear view of their purchase history.\\
+\textbf{Fit Criterion:} The system should present the user's purchase history accurately and in a user-friendly manner.\\
+
+\subsection{Formal Specification}
+In regards to identifying roles and functions for project:
+\begin{itemize}
+    \item \textbf{Employee must respond to users.}\\
+    For all users `a' who are employees, and for all users `b' who are regular users and send a message to `a':
+    \[
+    \begin{split}
+        \forall a \in \text{users} \quad (& a.\text{role} = \text{``employee"} \\ 
+        & \land \forall b \in \text{users} \quad (b.\text{role} = \text{``user"} \Rightarrow a.\text{respondTo}(b) \Rightarrow \text{true}))
+    \end{split}
+    \]
+
+    \item \textbf{Owners must respond to employees.}\\
```

Comment:
Sorry I mean remove everything that had to do with owners and employees so these first two as well. We should only have FRs from the context of users.

Reply:
I'll just remove it myself since I need to add the traceability matrix anyways.